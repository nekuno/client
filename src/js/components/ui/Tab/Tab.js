import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './Tab.scss';
import * as ReactTabs from 'react-tabs';

export default class Tab extends Component {

    static propTypes = {
        columns : PropTypes.number.isRequired,
        selected: PropTypes.bool
    };

    render() {
        const {columns, selected, children} = this.props;
        const widthPercentage = 100 / columns;
        const className = selected ? styles.selected + ' ' + styles.tab : styles.tab;

        return (
            <ReactTabs.Tab className={className} style={{width: widthPercentage + '%'}}>
                {children}
            </ReactTabs.Tab>
        );
    }
}

Tab.tabsRole = 'Tab';