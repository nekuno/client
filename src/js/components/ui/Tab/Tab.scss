@import '../../../../scss/_partials/colors';

.tab {
  border-bottom: 2px solid transparent;
  color: $deep-purple;
  display: inline-block;
  font-size: 13px;
  font-weight: bold;
  position: relative;
  list-style: none;
  padding: 6px 0;
  cursor: pointer;
  text-align: center;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  &.selected {
    border-bottom: 2px solid $deep-purple;
  }
}
