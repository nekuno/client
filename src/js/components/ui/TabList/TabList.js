import React, { Component } from 'react';
import styles from './TabList.scss';
import * as ReactTabs from 'react-tabs';

export default class TabList extends Component {

    render() {
        const {children} = this.props;

        return (
            <ReactTabs.TabList className={styles.tabList}>
                {children}
            </ReactTabs.TabList>
        );
    }
}

TabList.tabsRole = 'TabList';