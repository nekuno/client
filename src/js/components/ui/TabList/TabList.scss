.tab-list {
  line-height: 25px;
  margin: 0;
  padding: 0;
}