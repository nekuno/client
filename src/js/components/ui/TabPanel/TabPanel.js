import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './TabPanel.scss';
import * as ReactTabs from 'react-tabs';

export default class TabPanel extends Component {

    static propTypes = {
        selected: PropTypes.bool,
    };

    render() {
        const {selected, children} = this.props;

        return (
            <ReactTabs.TabPanel className={styles.tabPanel} selected={selected}>
                {children}
            </ReactTabs.TabPanel>
        );
    }
}

TabPanel.tabsRole = 'TabPanel';