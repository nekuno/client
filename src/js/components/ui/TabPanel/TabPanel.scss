.tab-panel {
  font-size: 13px;
  padding: 0 10px;
}