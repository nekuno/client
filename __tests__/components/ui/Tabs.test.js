import React from 'react';
import TabList from '../../../src/js/components/ui/TabList/TabList.js';
import Tab from '../../../src/js/components/ui/Tab/Tab.js';
import TabPanel from '../../../src/js/components/ui/TabPanel/TabPanel.js';
import { Tabs } from 'react-tabs';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test Tabs component', () => {
    const tabs = shallow(
        <Tabs>
            <TabList>
                <Tab columns={2}>Foo</Tab>
                <Tab columns={2}>Bar</Tab>
            </TabList>
            <TabPanel>
                <div>Lorem ipsum.</div>
            </TabPanel>
            <TabPanel>
                <div>Etiam et elit ante</div>
            </TabPanel>
        </Tabs>
    );

    it('should be defined', () => {
        expect(Tabs).toBeDefined();
    });
    it('should render correctly', () => {
        expect(tabs).toMatchSnapshot();
    });
});
