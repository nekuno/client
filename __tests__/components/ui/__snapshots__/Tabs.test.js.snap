// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Tabs component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tabs
    defaultFocus={false}
    defaultIndex={null}
    forceRenderTabPanel={false}
    selectedIndex={null}
  >
    <TabList>
      <Tab
        columns={2}
      >
        Foo
      </Tab>
      <Tab
        columns={2}
      >
        Bar
      </Tab>
    </TabList>
    <TabPanel>
      <div>
        Lorem ipsum.
      </div>
    </TabPanel>
    <TabPanel>
      <div>
        Etiam et elit ante
      </div>
    </TabPanel>
  </Tabs>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TabList>
          <Tab
            columns={2}
          >
            Foo
          </Tab>
          <Tab
            columns={2}
          >
            Bar
          </Tab>
        </TabList>,
        <TabPanel>
          <div>
            Lorem ipsum.
          </div>
        </TabPanel>,
        <TabPanel>
          <div>
            Etiam et elit ante
          </div>
        </TabPanel>,
      ],
      "className": "react-tabs",
      "focus": false,
      "forceRenderTabPanel": false,
      "onSelect": [Function],
      "selectedIndex": 0,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Tab
              columns={2}
            >
              Foo
            </Tab>,
            <Tab
              columns={2}
            >
              Bar
            </Tab>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Foo",
              "columns": 2,
            },
            "ref": null,
            "rendered": "Foo",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Bar",
              "columns": 2,
            },
            "ref": null,
            "rendered": "Bar",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div>
            Lorem ipsum.
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem ipsum.",
          },
          "ref": null,
          "rendered": "Lorem ipsum.",
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div>
            Etiam et elit ante
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Etiam et elit ante",
          },
          "ref": null,
          "rendered": "Etiam et elit ante",
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TabList>
            <Tab
              columns={2}
            >
              Foo
            </Tab>
            <Tab
              columns={2}
            >
              Bar
            </Tab>
          </TabList>,
          <TabPanel>
            <div>
              Lorem ipsum.
            </div>
          </TabPanel>,
          <TabPanel>
            <div>
              Etiam et elit ante
            </div>
          </TabPanel>,
        ],
        "className": "react-tabs",
        "focus": false,
        "forceRenderTabPanel": false,
        "onSelect": [Function],
        "selectedIndex": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Tab
                columns={2}
              >
                Foo
              </Tab>,
              <Tab
                columns={2}
              >
                Bar
              </Tab>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Foo",
                "columns": 2,
              },
              "ref": null,
              "rendered": "Foo",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Bar",
                "columns": 2,
              },
              "ref": null,
              "rendered": "Bar",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div>
              Lorem ipsum.
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lorem ipsum.",
            },
            "ref": null,
            "rendered": "Lorem ipsum.",
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div>
              Etiam et elit ante
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Etiam et elit ante",
            },
            "ref": null,
            "rendered": "Etiam et elit ante",
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
